gin.go

```diff
@@ -394,6 +398,25 @@ func (engine *Engine) parseTrustedProxies() error {
 	return err
 }
 
+// updateRouteTree do update to the route tree recursively
+func updateRouteTree(n *node) {
+	n.path = strings.ReplaceAll(n.path, escapedColon, colon)
+	n.fullPath = strings.ReplaceAll(n.fullPath, escapedColon, colon)
+	n.indices = strings.ReplaceAll(n.indices, backslash, colon)
+	if n.children != nil {
```

Comment:
[Reverse the condition return early.](https://szymonkrajewski.pl/why-should-you-return-early/)